scripts/ci/check_private_python_proxy_health.py

```
@@ -0,0 +1,572 @@
+import argparse
+from dataclasses import dataclass
+import json
+import os
+from pathlib import Path
+import re
+import socket
+import ssl
+import sys
+from typing import Any, Iterable, Sequence
+from urllib.error import HTTPError, URLError
+from urllib.parse import ParseResult, quote, urlparse
```

Flake8 (7.3.0)

[error] 'urllib.parse.ParseResult' imported but unused

(F401)